tests/test_app.py

```diff
@@ -70,3 +84,38 @@ def test_call(self):
         response = func(request)
         assert response.status == '200 OK'
         assert response.text == '9'
+
+    @pytest.mark.skipif(sys.version_info.major >= 3, reason="Requires Python 2")
```

Comment:
Can you define the following on the top:

    py2_only = pytest.mark.skipif(sys.version_info.major >= 3, reason="Requires Python 2")

And use:

    @py2_only
    def test_generate_signature(self):
        ...